### PR TITLE
notification integration test

### PR DESCRIPTION
**Pull request by oriyomibadmus**

### CODE REVIEW

**Comment by chungthuang** [comment]
There's a conflict in the integration test database. You might need to override with your version and see if that brakes any tests.

File: integration-test/collection.json [hunk 1]
```diff
@@ -154,8 +157,98 @@
 				}
 			},
 			"response": []
+		},
+
+		{
+			"name": "notification",
```
Comment (becevka) [inline integration-test/collection.json:163]: This test checks for the notification record existence in database and whether the response from API corresponds, the schema, which is good, but it tests the general function of Strapi. 
I noticed that both Idea-Card and Comment now have lifecycle hooks to create notification when idea/comment are added. I would suggest additional tests to add idea and verify that notification is added successfully, add a comment and verify that notification is added as expected.

File: integration-test/collection.json [hunk 2]
```diff
@@ -154,8 +157,98 @@
...
+							"                }",
+							"            },",
+							"            \"required\": [\"pagination\"]",
+							"        }",
+							"    },",
+							"    \"required\": [\"data\", \"meta\"]",
+							"};",
+							"",
+							"pm.test('Test Notification', function () {",
+							"    var result = tv4.validateResult(pm.response.json(), schema);",
```
Comment (becevka) [inline integration-test/collection.json:221]: Does it validate specific expected values?

File: integration-test/collection.json [hunk 3]
```diff
@@ -154,8 +157,136 @@
...
+							"                \"extraInfo\": \"tesst\",\r",
+							"                \"status\": \"submitted\",\r",
+							"                \"involveLevel\": \"none\",\r",
+							"                \"ideaOwner\": \"integration-test\"\r",
+							"            }\r",
+							"        })\r",
+							"    }\r",
+							"}, function (err, response) {\r",
+							"    pm.expect(err).to.be.null;\r",
+							"    pm.expect(response.code).to.be.oneOf([200, 201]);\r",
```
Comment (chungthuang) [inline integration-test/collection.json:193]: I think we only return 200 status code.

Comment (oriyomibadmus) [reply integration-test/collection.json:193]: Okay

File: src/api/notification/content-types/notification/schema.json
Comment (chungthuang) [inline src/api/notification/content-types/notification/schema.json:1]: This is already created in the main branch https://github.com/dev-launchers/strapiv4/blob/main/src/api/notification/content-types/notification/schema.json. Maybe you need to rebase on the main branch?

File: src/api/comment/content-types/comment/lifecycles.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
+module.exports = {
```
Comment (becevka) [inline src/api/comment/content-types/comment/lifecycles.js:1]: Strange, that this file shows as new, while we have it already in main 🤔

Comment (becevka) [reply src/api/comment/content-types/comment/lifecycles.js:1]: Oh, I see, there are multiple conflicts. Please make sure you have updated your branch with the latest main

File: src/api/notification/content-types/notification/schema.json
Comment (zar92) [reply src/api/notification/content-types/notification/schema.json:1]: This is solved

File: src/api/comment/content-types/comment/lifecycles.js [hunk 5]
```diff
@@ -1,16 +1,16 @@
...
-    console.log(event.result)
 
-    const { id: id, idea_card: idea, author: commentAuthor, text: commentText, createdAt: timeCreated } = event.result;
+    const { idea_card: idea, author: commentAuthor, text: commentText, createdAt: timeCreated } = event.params.data;
+    const commentId = event.result.id
     
     await strapi.entityService.create('api::notification.notification', {
       data: {
-        Title: commentAuthor + " commented on " + idea.ideaName,
+        Title: commentAuthor + " commented on ",
```
Comment (becevka) [inline src/api/comment/content-types/comment/lifecycles.js:10]: @oriyomibadmus Could you please add what author commented on?

File: src/api/notification/content-types/notification/schema.json [hunk 6]
```diff
@@ -16,22 +16,18 @@
...
       "type": "datetime"
     },
-    "Collection": {
-      "type": "enumeration",
-      "enum": [
-        "IdeaCard",
-        "Comment"
-      ]
+    "entityType": {
+      "type": "string"
```
Comment (becevka) [inline src/api/notification/content-types/notification/schema.json:28]: @oriyomibadmus Maybe better to keep the entityType to be enumeration?

**Review by becevka — APPROVED** [review]

**Pull request merged**